yoga/Yoga.c

```diff
@@ -781,14 +772,46 @@ static void YGPrintNumberIfNotUndefinedf(const char *str, const float number) {
 
 static void YGPrintNumberIfNotUndefined(const char *str, const YGValue *const number) {
   if (number->unit != YGUnitUndefined) {
-    YGLog(YGLogLevelDebug,
-          "%s: %g%s, ",
-          str,
-          number->value,
-          number->unit == YGUnitPoint ? "pt" : "%");
+    if (number->unit == YGUnitAuto) {
+      YGLog(YGLogLevelDebug, "%s: 'auto', ", str);
+    } else {
+      const char *unit = number->unit == YGUnitPoint ? "pt" : "%";
+      YGLog(YGLogLevelDebug, "%s: %g%s, ", str, number->value, unit);
+    }
+  }
+}
+
+static void YGPrintNumberIfNotZero(const char *str, const YGValue *const number) {
+  if (!YGFloatsEqual(number->value, 0)) {
+    YGPrintNumberIfNotUndefined(str, number);
   }
 }
 
+#define YG_STRING(s) #s
+
+#define YG_LOG_PRINT_NONZERO_EDGE(type, edge)           \
+  \
+YGPrintNumberIfNotZero(YG_STRING(type) YG_STRING(edge), \
+                       YGComputedEdgeValue(node->style.type, YGEdge##edge, &YGValueZero));
+
+#define YG_LOG_PRINT_NONZERO_EDGES(type)                                                     \
+  \
+if(YGFourValuesEqual(node->style.type)) {                                                    \
+    YGPrintNumberIfNotZero(YG_STRING(type),                                                  \
+                           YGComputedEdgeValue(node->style.type, YGEdgeLeft, &YGValueZero)); \
+  \
+}                                                                                       \
+  else {                                                                                     \
+    YG_LOG_PRINT_NONZERO_EDGE(type, Left);                                                   \
+    YG_LOG_PRINT_NONZERO_EDGE(type, Right);                                                  \
+    YG_LOG_PRINT_NONZERO_EDGE(type, Left);                                                   \
+    YG_LOG_PRINT_NONZERO_EDGE(type, Top);                                                    \
+    YG_LOG_PRINT_NONZERO_EDGE(type, Bottom);                                                 \
+    YG_LOG_PRINT_NONZERO_EDGE(type, Start);                                                  \
+    YG_LOG_PRINT_NONZERO_EDGE(type, End);                                                    \
+  \
+}
+
 static bool YGFourValuesEqual(const YGValue four[4]) {
```

Comment:
This function doesn't even handle start & end or vertical & horizontal so would love it fixes for that were included here as well 👍